[Deps] Update Dependency Analysis to 2.8.0 [`++`]

ParaskP7 opened the pull request:
<!-- Remember about a good descriptive title. -->

### Description
<!-- Take the time to write a good summary. Why is it needed? What does it do? When fixing bugs try to avoid just writing “See original issue” – clarify what the problem was and how you’ve fixed it. -->

This PR updates Dependency Analysis to [2.8.0](https://github.com/autonomousapps/dependency-analysis-gradle-plugin/blob/main/CHANGELOG.md#version-280), which is a major update from [1.28.0](https://github.com/autonomousapps/dependency-analysis-gradle-plugin/blob/main/CHANGELOG.md#version-1280), and thus was done via [2.0.2](https://github.com/autonomousapps/dependency-analysis-gradle-plugin/blob/main/CHANGELOG.md#version-202). (93df40913fb9315c153bf4cb70dc84009596f8e5)

> [!NOTE]
> In addition to the update itself, a couple more improvements are added to this change:
> 1. A Slack notification is posted for successful build too. (bcff0499635a18293d0c0f2511fad63118f131c0)
> 2. The dependency cache is now being restored as part of running this schedule build, just like with normal builds. (e16605aba701ab3154480152f55fb70b9f68f2de)

---

### Testing information
<!-- This is your opportunity to break out individual scenarios that need testing (when necessary) and/or include a checklist for the reviewer to go through. Consider documenting the following from your own completed testing: devices used, alternate workflows, edge cases, affected areas, critical flows, areas not tested, and any remaining unknowns. Provide feedback on this new section of the PR template through Sept 30, 2024 to Apps Quality; additional context here: p91TBi-b8z-p2#comment-12036 -->

1. Tooling:
    - Run the `./gradlew buildHealth` task and verify that there exists an `unused` related advise within both reports, JSON and txt included:
        - `build-health-report.json` -> Search for `unusedCount`
        - `build-health-report.txt` -> Search for `unused`
    - Check the manually triggered [scheduled build](https://buildkite.com/automattic/woocommerce-android/builds/26927) and verify that there exists an `unused` related advise within both reports, JSON and txt included:
        - `build-health-report.json` -> Search for `unusedCount`
        - `build-health-report.txt` -> Search for `unused`
2. Notifications: (`#android-core-notifs`)
    - Verify that the notification change works as expected:
        - On success, notifying that: `Dependency analysis succeeded.` (p1738256615099189-slack-C0787KRDYDC)
        - On failure, notifying that: `Dependency analysis failed.` (not tested, but should behave similarly to ☝️)
3. Dependency Cache:
    - Verify that the dependency cache is being restored as expected, and also, that it works as expected:
        - [Restoring Gradle dependency cache...](https://buildkite.com/automattic/woocommerce-android/builds/26927#0194b80c-d226-493d-a384-829bf2636fd4/188-189)
        - [Build Scan](https://buildkite.com/automattic/woocommerce-android/builds/26927#0194b80c-d226-493d-a384-829bf2636fd4/202-436) -> [Network Activity](https://scans.gradle.com/s/t6sbb57umjote/performance/network-activity)
5. Further Testing (`Optional`):
    - Smoke test the app, and verify everything is working as expected.
    - Try to test for build time increases both, locally and on CI (for example, you could see a [previous such scheduled build](https://buildkite.com/automattic/woocommerce-android/builds/26834) and compare the diff[^1]).
    - Check [build environment changes](https://github.com/woocommerce/woocommerce-android/pull/13424#issuecomment-2625247881).

[^1]: Note: You will actually notice a significant diff, the job used to take [8m 56s](https://buildkite.com/automattic/woocommerce-android/builds/26834#01949feb-32dc-4f6a-bfe2-9e8a44ba021f) previously, and now it is taking [32m 11s](https://buildkite.com/automattic/woocommerce-android/builds/26927#0194b80c-d226-493d-a384-829bf2636fd4), an increase of more than `20m`. However, this is actually expected with [this issue](https://github.com/autonomousapps/dependency-analysis-gradle-plugin/issues/1186) still persisting, which causes trouble for some project. Having said that, I don't think we should block the updates anymore, especially given the fact that this job is running once a week.

---

- [x] I have considered if this change warrants release notes and have added them to `RELEASE-NOTES.txt` if necessary. Use the "[Internal]" label for non-user-facing changes.

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:

- [x] The PR is small and has a clear, single focus, or a valid explanation is provided in the description. If needed, please request to split it into smaller PRs.
- [x] Ensure Adequate Unit Test Coverage: The changes are reasonably covered by unit tests or an explanation is provided in the PR description.
- [x] Manual Testing: The author listed all the tests they ran, including smoke tests when needed (e.g., for refactorings). The reviewer confirmed that the PR works as expected on big (tablet) and small (phone) in case of UI changes, and no regressions are added.

<!-- Pull request guidelines: https://github.com/woocommerce/woocommerce-android/blob/develop/docs/pull-request-guidelines.md -->

ParaskP7 committed:
Deps: Update dependency analysis to 2.0.2

Release Notes: https://github.com/autonomousapps/
dependency-analysis-gradle-plugin/blob/main/CHANGELOG.md#version-202

------------------------------------------------------------------------

With the Dependency Analysis 2.0 update, a behavioral change made the
plugin to no longer automatically apply itself to every subproject. The
extra changes in this commit brings back this functionality,
applying the plugin explicit on each module.

There also exists another way of dealing with this, to use the new
'com.autonomousapps.build-health' plugin instead, but for the time being
let's not introduce an additional change, and just fix the main problem
for now, which is the plugin not reporting anything after its major 2.0
update.

For more info see: https://dev.to/autonomousapps/

ParaskP7 committed:
Deps: Update dependency analysis to 2.8.0

Release Notes: https://github.com/autonomousapps/
dependency-analysis-gradle-plugin/blob/main/CHANGELOG.md#version-280

ParaskP7 committed:
CI: Notify on slack channel when dependency analysis fails, and succeeds

This change follows a similar approach already established on the
'Dependency Cache' side of thing and workflow, ultimately making the
success path explicit and more visible.

ParaskP7 committed:
CI: Add restore cache step to dependency analysis

dangermattic commented:
<!--
  0 Errors
  0 Warnings
  1 Message: This PR is still a Draft: some...
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

atorresveiga commented:
Hi @ParaskP7 . when testing the tooling, I got this error:

![image](https://github.com/user-attachments/assets/21e78a98-b92a-4529-aa0e-5a93ced3c347)

Is this expected? ☝️ 

I smoke-tested the app, and it is working 🎉 

Checked the existence of `unusedCount` and `unused`

The code looks good to me

atorresveiga reviewed (approved):
LGTM! :shipit:

ParaskP7 committed:
Merge branch 'trunk' of github.com:woocommerce/woocommerce-android into deps/update-dependency-analysis-to-2.8.0

ParaskP7 committed:
Deps: Update dependency analysis to 2.0.2 (new modules)

This is (kind of) a continuation of this initial
93df40913fb9315c153bf4cb70dc84009596f8e5 change, which didn't include
the 'fluxc' and 'login' related modules since they weren't vendored into
'trunk' just yet.

Yesterday, the vendoring process got completed and after getting the
latest changes from 'trunk' those newly added 'fluxc' and 'login'
related modules need to have the dependency analysis plugin applies to
them as well.

ParaskP7 commented:
Thanks for the review @atorresveiga ! 🙇 ❤️ 🚀 

> ...I got this error:
> Is this expected? ☝️

Hmmm, no error is expected, but from the screenshot I can't see the exception to which the build failed, thus I am not sure what's really happening there... 🤔 

Did you manage to replicate this error when running `./gradlew buildHealth` locally, if yes, can you share the exception? 🙏 

---

FYI: Because these #13446 + #13447 + #13304 vendor related PRs got merged yesterday, them adding `fluxc` + `login` modules into the project, I had to merge `trunk` into this PR and add dependency analysis on those newly added modules as well. Thus, I just triggered another manual [schedule build](https://buildkite.com/automattic/woocommerce-android/builds/27036), will wait for that to complete and then move forward. Cc @wzieba

ParaskP7 commented:
Any extra thoughts on [this](https://github.com/woocommerce/woocommerce-android/pull/13424#issuecomment-2633222109) @atorresveiga before I merge it to `trunk`? 🙏 

PS: @wzieba could you briefly look into that as well, maybe give me a quick 👍 too, before me merging this. 🙏

atorresveiga commented:
> Any extra thoughts on https://github.com/woocommerce/woocommerce-android/pull/13424#issuecomment-2633222109 @atorresveiga before I merge it to trunk? 🙏

@ParaskP7  After the merge with trunk, the build displays a success message; I guess it needed an update in my local configuration.

![image](https://github.com/user-attachments/assets/bae15d9b-62cb-4891-b0ec-155ab8f24161)

ParaskP7 commented:
> @ParaskP7 After the merge with trunk, the build displays a success message; I guess it needed an update in my local configuration.

Awesome, thanks for the clarification on that @atorresveiga , merging then! 🚀 

PS: Thanks to you too @wzieba ! 🙇
